pkg/mount/mountutil.go

```diff
@@ -333,3 +339,27 @@ func fileExists(file string) bool {
 	}
 	return true
 }
+
+// mergeDeviceMountAttrs merges the second mountattr into the first. The merge is
+// performed as follows:
+// 1. If the DevPath of the first mountattr is empty, then it is set to the DevPath of
+// the second mountattr
+// 2. If the FileSystem of the first mountattr is empty, it is set to the FileSystem of
+// the second mountattr provided that the DevPaths of both the mountattrs match
+// 3. The MountPoint(s) of the second mountattr are appended to first's only if the
+// DevPath and the FileSystem of both the mountattrs match and the FileSystem of first
+// mountattr is non-empty (!= "")
+func mergeDeviceMountAttrs(ma *DeviceMountAttr, mb *DeviceMountAttr) {
+	if ma.DevPath == "" {
```

Comment:
Why is this check needed? If `DevPath` is empty we can return empty without merging

Reply:
This is required in cases when `getPartitionName` is used. In this case, the DevPath of `ma` will be empty but DevPath of `mb` will be set by `getPartitionName`. So we need to merge that info into the result which is `ma`.

Reply:
ok. got it.

Reply:
Can you add test cases when both the functions are called so that we dont break any existing functionality.

Reply:
sure. I saw a to-do in the tests file for adding tests for `getPartitionName`. I'll add those tests in a separate PR and then update this one (I prefer keeping separate changes in separate PRs 🙈 )

Reply:
Thats good.